perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -657,4 +666,51 @@ public int getGroupsCount(PerunSession sess) throws InternalErrorException {
 			throw new InternalErrorException(e);
 		}
 	}
+
+	@Override
+	public List<Integer> getRelatedGroupsIds(PerunSession sess, int groupId) {
+		return jdbc.queryForList("SELECT result_gid FROM groups_groups WHERE operand_gid=" + groupId, Integer.class);
+	}
+
+	@Override
+	public void removeGroupUnion(PerunSession sess, Group resultGroup, Group operandGroup) throws InternalErrorException {
+		if (0 == jdbc.update("DELETE FROM groups_groups WHERE result_gid = ? AND operand_gid = ?",
+				resultGroup.getId(), operandGroup.getId())) {
+			throw new InternalErrorException("There is no relation between " + resultGroup + " and " + operandGroup);
+		}
+	}
+
+	@Override
+	public void removeResultGroupRelations(PerunSession sess, Group resultGroup) {
+		jdbc.update("DELETE FROM groups_groups WHERE result_gid = ?", resultGroup.getId());
+	}
+
+	@Override
+	public void saveGroupRelation(PerunSession sess, Group resultGroup, Group operandGroup, boolean parentFlag) throws InternalErrorException {
+		jdbc.update("INSERT INTO groups_groups(result_gid, operand_gid, created_at, created_by, " +
+						"modified_at, modified_by, parent_flag) VALUES(?,?," + Compatibility.getSysdate() + ",?," + Compatibility.getSysdate() + ",?,?)",
+				resultGroup.getId(), operandGroup.getId(), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getActor(), parentFlag);
+	}
+	
+	public boolean isRelationRemovable(PerunSession sess, Group resultGroup, Group operandGroup) {
+		return 1 > jdbc.queryForInt("SELECT parent_flag FROM groups_groups WHERE result_gid=? AND operand_gid=?", 
+				resultGroup.getId(), operandGroup.getId());
+	}
+
+	@Override
+	public boolean isRelationBetweenGroups(Group group1, Group group2) {
+		return 1 <= jdbc.queryForInt("SELECT count(1) FROM groups_groups WHERE (result_gid = ? AND operand_gid = ?) OR (result_gid = ? AND operand_gid = ?)",
+				group1.getId(), group2.getId(), group2.getId(), group1.getId());
+	}
+
+	@Override
+	public boolean isOneWayRelationBetweenGroups(Group resultGroup, Group operandGroup) {
+		return 1 <= jdbc.queryForInt("SELECT count(1) FROM groups_groups WHERE result_gid = ? AND operand_gid = ?",
+				resultGroup.getId(), operandGroup.getId());
+	}
+
+	@Override
+	public List<Integer> getGroupRelations(PerunSession sess, int groupId) {
+		return jdbc.queryForList("SELECT result_gid FROM groups_groups WHERE operand_gid=" + groupId, Integer.class);
```

Comment:
:poop: Always use prepared statements (Substitute variables for '?' in SQL query)!